app/controllers/dams_resource_controller.rb

```diff
@@ -229,14 +213,24 @@ def osf_push
     end
 
     if document.valid?
-      replace_config_file ('share_notify.yml')
-      api = ShareNotify::API.new
-      api.post(document.to_share.to_json)
-      copy_config_file ('share_notify.yml') if !Rails.env.production?
-      redirect_to dams_collection_path(params[:id]), notice: "Your record has been pushed to OSF Share Staging area."
+      @headers = {'Authorization' => "Token #{share_config.fetch('token', nil)}", 
+                 'Content-Type'  => 'application/json'
+                }
+      @route = "#{share_config.fetch('host', 'https://staging.osf.io/')}api/v1/share/data"
```

Comment:
Do we need a default here from the `#fetch` call? It looks like the config is already providing defaults.